components/RestPasswordForm/ResetPasswordForm.tsx

```diff
@@ -0,0 +1,192 @@
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+import { Button } from '@/components/Button/Button';
+import { Input } from '@/components/Input/Input';
+import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { JSX, useState } from 'react';
+import { Control, SubmitHandler, useForm, useWatch } from 'react-hook-form';
+import toast from 'react-hot-toast';
+import { z } from 'zod';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '../Form/Form';
+import { useAuthContext } from '@/context/AuthContextProvider';
+
+const ResetPasswordSchema = z.object({
+  oldPassword: z.string().min(1, { message: 'Please enter your old password' }),
+  newPassword: z
+    .string()
+    .min(1, { message: 'Please enter your new password' })
+    .min(6, { message: 'Password must be at least 6 characters' }),
+});
+
+type ResetPasswordSchemaType = z.infer<typeof ResetPasswordSchema>;
+
+/**
+ * Display update user email form
+ * @returns {JSX.Element} The rendered update user email form.
+ */
+const ResetPasswordForm = (): JSX.Element => {
+  const [isUpdating, setIsUpdating] = useState<boolean>(false);
+  const { logoutAccount } = useAuthContext();
+
+  const form = useForm<ResetPasswordSchemaType>({
+    resolver: zodResolver(ResetPasswordSchema),
+    defaultValues: {
+      oldPassword: '',
+      newPassword: '',
+    },
+  });
+
+  const oldPassword: string = useWatch({
+    control: form.control,
+    name: 'oldPassword',
+  });
+
+  const newPassword: string = useWatch({
+    control: form.control,
+    name: 'newPassword',
+  });
+
+  /**
+   * A function that handles form submission.
+   * @param {ResetPasswordSchemaType} data - The data submitted in the form.
+   * @returns {Promise<void>} Promise that resolves after form submission is processed.
+   */
+  const onSubmit: SubmitHandler<ResetPasswordSchemaType> = async (
+    data: ResetPasswordSchemaType,
+  ): Promise<void> => {
+    const { oldPassword, newPassword } = data;
+    setIsUpdating(true);
+    try {
+      await resetPassword({
+        newPassword,
+        oldPassword,
+      });
+      toast.custom(
+        <Alert
+          variant={AlertVariants.Success}
+          message="You have successfully updated your password."
+        />,
+      );
+
+      form.reset({ oldPassword: '', newPassword: '' });
+    } catch (error) {
+      console.error('Password Update Failed', error);
+      toast.custom(
+        <Alert
+          variant={AlertVariants.Error}
+          message="Password Update Failed!"
+        />,
+      );
+    } finally {
+      setIsUpdating(false);
+    }
+  };
+
+  const isDisabled = oldPassword === '' || newPassword === '' || isUpdating;
+
+  return (
+    <Form {...form}>
+      <form
+        id="password-input-container"
+        className="border border-border rounded-lg w-full bg-border/40 p-6"
```

Comment:
```suggestion
        className="border border-border rounded-radius w-full bg-border/40"
```

- We have a `radius` CSS variable for `rounded` :)

Reply:
@ryandotfurrer ignoring per your request to ignore in Discord

Reply:
For UX request, let's create a ticket for them. Segregating Functional from UX.